R/html_document_base.R

```diff
@@ -184,3 +178,13 @@ html_document_base <- function(smart = TRUE,
   )
 }
 
+extract_preserve_chunks <- function(input_file, extract = extractPreserveChunks) {
+  # The input file is converted to UTF-8 from its native encoding prior
+  # to calling the preprocessor (see ::render)
+  input_str <- readLines(input_file, warn = FALSE, encoding = "UTF-8")
+  preserve <- extract(input_str)
+  if (!identical(preserve$value, input_str))
+    writeLines(enc2utf8(preserve$value), input_file, useBytes = TRUE)
```

Comment:
That was intentional. We should make sure `input_file` is always encoded in UTF-8, and `enc2utf8()` here should not be harmful. If `preserve$value` was converted to the native encoding for some reason on Windows, we would be doing the wrong thing by `writeLines(preserve$value, input_file, useBytes = TRUE)`.